### DISCUSSION TITLE
loki log explore add progress bar

### DISCUSSION THREAD

**Discussion by liguozhong** [grafana/grafana#61598] [Explore]
**Is your feature request related to a problem? Please describe.**
Because the execution of `metrics query` in loki is very slow, some take 3 minutes, and some take longer.
for example，We have two clusters, one with a large amount of data and one with a small amount of data. In a cluster with a small amount of data, logql can return within 5 seconds. In a cluster with a large amount of data, logql takes 3 minutes to return.

We need a better user experience to allow users to be more patient while waiting.
So we need to provide an asynchronous http API for log explore to query the logql execution progress percentage.

**Describe the solution you'd like**
1: Add a field `uuid` to the return value of `/loki/api/v1/query_range` of each logql to mark itself.
2: add a `/loki/api/v1/process/<uuid>` new http api for grafana ui.

When loki explore is waiting for the corresponding process, add a progress prompt box, for example: 
![image](https://user-images.githubusercontent.com/9583245/212833598-a55ec044-e634-4781-b54d-545707846f95.png)
![image](https://user-images.githubusercontent.com/9583245/212833702-9b592c45-65b4-4cdb-a7b7-5120fff94f55.png)
[xxxxxxxx--]80%

**Describe alternatives you've considered**
streaming http response, but it is difficult.

**Additional context**
loki slow response.
![image](https://user-images.githubusercontent.com/9583245/212833143-d3ccc228-64bc-41c4-adfa-720d09a8bc0f.png)

**Comment by liguozhong** [comment]
https://github.com/grafana/grafana/issues/61769

**Comment by gelicia** [comment]
Quoting the linked issue

> Unfortunately there's no way to display a progress bar as we just wait for a response that will take an unknown amount of time to be resolved. For those who will enable and use query splitting (or chunking), we might add a progress indicator, but right now it's not a priority https://github.com/grafana/grafana/issues/65104

The issue linked there is about query splitting and was resolved. I think we have been able to accomplish everything we can for this feature request for now. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if there is more to this that was not addressed.
